src/bluetooth-fw/nimble/nimble_store.c

```diff
@@ -43,125 +44,147 @@ static bool prv_nimble_store_find_sec_cb(ListNode *node, void *data) {
   return ble_addr_cmp(&s->value_sec.peer_addr, &key_sec->peer_addr) == 0;
 }
 
-static BleStoreValue *prv_nimble_store_find_sec(const struct ble_store_key_sec *key_sec) {
+static ListNode **prv_find_sec_list_for_obj_type(const int obj_type) {
+  switch (obj_type) {
+    case BLE_STORE_OBJ_TYPE_OUR_SEC:
+      return (ListNode **)&s_our_value_secs;
+    case BLE_STORE_OBJ_TYPE_PEER_SEC:
+      return (ListNode **)&s_peer_value_secs;
+    default:
+      assert(0);
```

Comment:
`return NULL`, prv_nimble_store_find_sec will just end up NULLing as well and error propagated/handled without crashing.

Reply:
It'd cause a problem (leaked new item) in `prv_nimble_store_upsert_sec` further down. Seems better to fail loudly if there's some value type we're not supporting but should?

Reply:
If you think we still shouldn't assert in this case, I can rework that case too to return an error code.

Reply:
it's ok, I always prefer to propagate errors unless it's explicitly a programming error (here it's really in the limbo). maybe `PBL_ASSERT` though?

Reply:
Ah, yep, switched the assert.